enzy_htp/analysis/rmsd.py

```diff
@@ -0,0 +1,120 @@
+    return rmsd_of_region(
+        structure_ensemble=structure_ensemble, 
+        interface=interface,
+        reference_structure=reference_structure, 
+        mask_region=mask_region, 
+        ca_only=ca_only
+    )
+
+def rmsd_of_region(structure_ensemble: StructureEnsemble, interface: str = "amber", reference_structure: Structure = None, mask_region: List[Residue] = list(), ca_only: bool = True) -> float:
+    """Get RMSD value from MD simulation result with a mask region (a list of residues).
+    
+    Args:
+        structure_ensemble (StructureEnsemble): A collection of different geometries of the same enzyme structure.
+        interface (str, optional): The interface for calculation. Default amber.
+        reference_structure (Structure, optional): The reference structure, which should have the same sequences and ligangs as the structure_ensemble. Default: Average structure.
+        mask_region (List[Residue], optional): A list of Residues to calculation RMSD value. If the list is empty, then all residues are selected.
+        ca_only (bool, optional): Indicate if only C-alpha are included in RMSD calculation; otherwise all atoms except hydrogens are included. Default True.
+    """
+    interface = interface.lower()
+    mask_pattern = compose_mask_pattern(mask_region=mask_region, interface=interface, ca_only=ca_only)
+    return rmsd_with_pattern(
+        structure_ensemble=structure_ensemble, 
+        interface=interface,
+        reference_structure=reference_structure, 
+        mask_pattern=mask_pattern
+    )
+
+def rmsd_with_pattern(structure_ensemble: StructureEnsemble, interface: str = "amber", reference_structure: Structure = None, mask_pattern: str = str()) -> float:
+    """
+    mvp function for RMSD calculation
+    
+    Args:
+        structure_ensemble (StructureEnsemble): A collection of different geometries of the same enzyme structure.
+        interface (str, optional): The interface for calculation. Default amber.
+        reference_structure (Structure, optional): The reference structure, which should have the same sequences and ligangs as the structure_ensemble. Default: Average structure.
+        mask_pattern (str): A pymol-formatted selection string which defines the region for calculating RMSD value.
+    """
+    interface = interface.lower()
+    if (interface == "amber"):
+        return ACCEPTED_INTERFACES[interface].get_rmsd(traj_path=structure_ensemble.coordinate_list, prmtop_path=structure_ensemble.topology_source_file, mask_pattern=mask_pattern)
+    elif (interface == "pymol"):
+        return ACCEPTED_INTERFACES[interface].get_rmsd(structure_ensemble=structure_ensemble, reference_structure=reference_structure, mask_pattern=mask_pattern)
+
+    elif (interface not in ACCEPTED_INTERFACES.keys()):
+        raise KeyError(f"Unacceptable interface value. Please choose from {' and '.join(ACCEPTED_INTERFACES.keys())}")
```

Comment:
The if flow here is confusing. The use of the mapper and the if flow is repetative. (significantly reduced the readablity) Please use the mapper only. Reference binding.py.

Reply:
Resolved.